src/main/java/ch/uzh/ifi/hase/soprafs22/lobby/LobbyManager.java

```diff
@@ -20,7 +20,7 @@
 
 public class LobbyManager implements ILobbyManager {
 
-    private volatile static LobbyManager uniqueInstance;
+    private static final LobbyManager uniqueInstance = new LobbyManager();
```

Comment:
Was that a code smell?
Before you had the (almost) textbook like Singleton applied, why did you change it?

Reply:
Yes, sonarCloud said, if I remember correctly that volatile only works with atomic types or something like that. I then read online that the method before is not completely thread save. The new way is the thread safest option as far as I know and read but you always create an instance even if you might not need it. In our case that shouldn't be a problem as the lobby manager is crucial and needs to be generated anyways. The old way would probably also work but one less smell I guess (but maybe it will grumble about that too 😢)

Reply:
Ah I remember, that was one of the drawbacks of Singletons, it's not thread safe.
Although it promises by design a single instance, multiple threads could create multiple Singletons.